basex/src/main/java/com/formulasearchengine/mathosphere/basex/TexQueryGenerator.java

```diff
@@ -11,81 +18,102 @@
+            .setUrl("http://drmf-latexml.wmflabs.org")
+            .setParams(updateParamFormat(this.getParams()));
+    private LaTeXMLServiceResponse serviceResponse;
+
+
+    List<NameValuePair> getParams() {
+        return params;
+    }
+
+    @NotNull
+    private ArrayList<NameValuePair> getDefaultParams() {
+        ArrayList<NameValuePair> pDefault = new ArrayList<>();
+        pDefault.add(new BasicNameValuePair("format", "xhtml"));
+        pDefault.add(new BasicNameValuePair("whatsin", "math"));
+        pDefault.add(new BasicNameValuePair("whatsout", "math"));
+        pDefault.add(new BasicNameValuePair("cmml", ""));
+        pDefault.add(new BasicNameValuePair("nodefaultresources", ""));
+        pDefault.add(new BasicNameValuePair("preload", "LaTeX.pool"));
+        pDefault.add(new BasicNameValuePair("preload", "article.cls"));
+        pDefault.add(new BasicNameValuePair("preload", "amsmath.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "amsthm.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "amstext.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "amssymb.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "eucal.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "[dvipsnames]xcolor.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "url.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "hyperref.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "mws.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "texvc"));
+        return pDefault;
+    }
+
+    void setParams(List<NameValuePair> params) {
+        this.params = params;
+        lateXMLConfig.setParams(updateParamFormat(params));
+    }
+
+    private Map<String, Object> updateParamFormat(List<NameValuePair> params) {
+        HashMap<String, Object> map = new HashMap<>();
+        for (NameValuePair pair : params) {
+            if (map.containsKey(pair.getName())) {
+                Object oSetting = map.get(pair.getName());
+                if (oSetting instanceof List) {
+                    ((List<String>) oSetting).add(pair.getValue());
+                } else {
+                    assert (oSetting instanceof String);
+                    map.put(pair.getName(), Lists.newArrayList(pair.getValue(), oSetting));
+                }
+            } else {
+                map.put(pair.getName(), pair.getValue());
+            }
+        }
+        return map;
+    }
+
+    String getLaTeXMLURL() {
+        return lateXMLConfig.getUrl();
+    }
+
+    Map getOb() {
+        HashMap<String, Object> hashMap = new HashMap();
+        hashMap.put("status_code", serviceResponse.getStatusCode());
+        hashMap.put("status", serviceResponse.getStatus());
+        hashMap.put("log", serviceResponse.getLog());
+        hashMap.put("result", serviceResponse.getResult());
+        return hashMap;
```

Comment:
I wanted to keep my old code and get the plain JSON response from the server. Neither that nor a deserialization of the response class to JSON was doable for me. So I finally gave up and ended up copying the old JSON annotations of the new class. I disliked that, and this is an extremely ugly hack. :-(